test/agentchat/test_conversable_agent.py

```diff
@@ -1263,6 +1263,29 @@ def test_messages_with_carryover():
     with pytest.raises(InvalidCarryOverType):
         agent1.generate_init_message(**context)
 
+def test_adding_duplicate_function_warning():
```

Comment:
Also add a unit test for `update_tool_signature` and `update_function_signature`

Reply:
@ekzhu it looks like to test update_tool_signature, we have to have an llm_config. I see in all the tests that there is no llm_config passed. Do you have an idea to get around this?

```AssertionError: To update a function signature, agent must have an llm_config
```

Reply:
@ekzhu Added! 
Tests run locally but fail in cicd because there is no OAI_CONFIG_LIST, how do we get around this?

Reply:
@ekzhu added, thanks!